core/src/main/java/hudson/model/Computer.java

```diff
@@ -370,7 +384,7 @@ public OfflineCause getOfflineCause() {
      */
     @Exported
     public String getOfflineCauseReason() {
-        if (offlineCause == null) {
+        if (temporarilyOfflineCause == null) {
```

Comment:
Not sure this replacement is correct. I would expect a similar behavior if there is non-temporary offline cause

Reply:
Theoretically it is ok. I looked up all places where this method is used and they usually meant to get the temporarily offline reason (not that this is a good way of checking if a previously call to setTemporarilyOffline with a certain message was done). Nevertheless I added another method to differentiate the 2 causes.